### PR TITLE
Dynamic frontend for the sidebar navigation Items

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

In this PR I've created a dynamic frontend that uses the new NavigationLinks table. This means that the sidebar is no longer hardcoded but can be configured from http://localhost:3000/admin/navigation_links. 

In addition to this PR there is also related docs which you can find https://app.gitbook.com/@forem/s/forem-admin-guide/~/drafts/-MJN9RF0nZtuN9YWy-IM/admin/navigation-links/@drafts

## Related Tickets & Documents
Follow up PR of  https://github.com/forem/forem/pull/9444 and https://github.com/forem/forem/pull/10662
https://github.com/forem/forem/issues/4661
https://github.com/forem/forem/issues/5475

## QA Instructions, Screenshots, Recordings

1. Open up a rails console
2. Run `NavigationLink.all`, ensure that the count matches the sidebar item count on http://localhost:3000/
3. Add, edit and delete items and ensure that it shows the updated changes on the homepage.

A view of the backend:
Screenshot 2020-10-12 at 11 46 38

A view of the corresponding frontend:
Screenshot 2020-10-12 at 11 46 22

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.forem.com
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -574,6 +574,12 @@ def create_if_doesnt_exist(klass, attribute_name, attribute_value)
 
 ##############################################################################
 
+seeder.create_if_none(ProfileField) do
```
Comment (citizen428) [inline db/seeds.rb:577]:
```suggestion
seeder.create_if_none(NavigationLink) do
```

File: lib/tasks/add_navigation_links.rake [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    listing_icon = File.read(Rails.root.join("app/assets/images/twemoji/listing.svg")).freeze
+    mic_icon = File.read(Rails.root.join("app/assets/images/twemoji/mic.svg")).freeze
+    camera_icon = File.read(Rails.root.join("app/assets/images/twemoji/camera.svg")).freeze
+    tag_icon = File.read(Rails.root.join("app/assets/images/twemoji/tag.svg")).freeze
+    bulb_icon = File.read(Rails.root.join("app/assets/images/twemoji/bulb.svg")).freeze
+    shopping_icon = File.read(Rails.root.join("app/assets/images/twemoji/shopping.svg")).freeze
+    heart_icon = File.read(Rails.root.join("app/assets/images/twemoji/heart.svg")).freeze
+    rainbowdev = File.read(Rails.root.join("app/assets/images/rainbowdev.svg")).freeze
+
+    NavigationLink.where(url: "#{base_url}/readinglist").first_or_create(name: "Reading List", icon: reading_icon,
```
Comment (citizen428) [inline lib/tasks/add_navigation_links.rake:25]: ** comment (non-blocking):** This is a very minor thing, but I find this very hard to read. I cases like this one I usually use this form:

```ruby
 NavigationLink.where(url: "#{base_url}/readinglist").first_or_create(
  name: "Reading List",
  icon: reading_icon,
  display_only_when_signed_in: true,
  position: 0
)
```

Yes, it uses up slightly more vertical space but that's generally not a problem and it's much easier to follow as it moves the attributes towards the beginning of the line.

Comment (Ridhwana) [reply lib/tasks/add_navigation_links.rake]: ✅

File: spec/system/homepage/user_visits_homepage_spec.rb [hunk 3]
```diff
@@ -106,24 +128,83 @@
       end
     end
 
-    describe "shop url" do
```
Comment (Ridhwana) [inline spec/system/homepage/user_visits_homepage_spec.rb:109]: The shop url is no longer hardcoded in the sidebar hence we've removed this test.

**Review by benhalpern — APPROVED** [review]
Works perfectly for me.

File: app/views/admin/navigation_links/_edit_navigation_link_modal.html.erb [hunk 4]
```diff
@@ -16,26 +16,7 @@
...
-            </div>
-            <div class="form-group">
-              <%= form.label :position %>
-              <%= form.number_field :position, class: "form-control" %>
-            </div>
-            <div class="form-group">
-              <%= form.check_box :display_only_when_signed_in %>
-              <%= form.label :display_only_when_signed_in %>
-            </div>
+            <%= render "form", form: form %>
```
Comment (juliannatetreault) [inline app/views/admin/navigation_links/_edit_navigation_link_modal.html.erb:19]: ❤️ this change!

**Review by juliannatetreault — APPROVED** [review]
I tested this out locally and worked as expected! Awesome work. ✨

**Pull request merged**